src/api/User/userRouter.ts

```diff
@@ -9,6 +9,9 @@ const User = new UserController();
 
 router.get("/:id", call(User.getUserDetails, (req, _res, _next) => [req.params.id]));
 
-router.post("/", call(User.createMainUser, (req, _res, _next) => [req.body]));
-    // tslint:disable-next-line: max-line-length
+router.post("/", call(User.createUserDetails, (req, _res, _next) => [req.body]));
+
+router.put("/:username", call(User.updateUser, (req, _res, _next) => [req.params.username, req.body.firstName]));
+
```

Comment:
User should be able to update all information except username so taking only firstName is wrong